index.bs

```diff
@@ -2334,6 +2309,121 @@ The [=default allowlist=] for this feature is <code>["self"]</code>.
 
 Note: If the document's origin is not allowed to use the <code>"xr-spatial-tracking"</code> feature policy any [=immersive sessions=] will be blocked, because all [=immersive sessions=] require some use of spatial tracking. [=Inline sessions=] will still be allowed, but restricted to only using the {{XRReferenceSpaceType/"viewer"}} {{XRReferenceSpace}}.
 
+
+Permissions API Integration {#permissions}
+---------
+
+The [[!permissions]] API provides a uniform way for websites to request permissions from users and query which permissions they have been granted. 
+
+The <dfn enum-value for="PermissionName">"xr"</dfn> [=powerful feature=]’s permission-related algorithms and types are defined as follows: 
+
+
+<dl>
+<dt>[=permission descriptor type=]</dt>
+<dd>
+
+<pre class="idl">
+dictionary XRPermissionDescriptor: PermissionDescriptor {
+  XRSession session;
+  sequence&lt;any&gt; requiredFeatures;
+  sequence&lt;any&gt; optionalFeatures;
+};
+</pre>
+
+{{PermissionDescriptor/name}} for {{XRPermissionDescriptor}} is {{PermissionName/"xr"}}.
+
+
+</dd>
+
+<dt>[=permission result type=]</dt>
+<dd>
+<pre class=idl>
+dictionary XRPermissionStatus: PermissionStatus {
+  sequence&lt;any&gt; granted;
+};
```

Comment:
This said, I wonder if we could delay this addition. Are there strong use cases for this? It would require implementations to be able to provide the list of enabled features without creating the session which may or may not an easy thing to do. If we don't have a strong use case, keeping the change small for now may be good?

Reply:
> would require implementations to be able to provide the list of enabled features without creating the session which may or may not an easy thing to do

I don't understand why? The spec already requires you to compute which features have been granted when you requestSession(), this just exposes that list.

Reply:
At the point of `requestSession()`, you already have an `XRSession`.

Reply:
No, this was _part of the point_  of this change, see https://github.com/immersive-web/webxr/issues/794 . It's important for apps to know what optional features were granted.

> Here, you should be able to check for permissions without a session.

This is already true, you don't need a session anymore.